framework/physics/TimeStepControllers/TimeStepController.cc

```diff
@@ -1,28 +1,115 @@
 #include "TimeStepController.h"
 
+#include <cmath>
+
 namespace chi_physics
 {
 
 chi::InputParameters TimeStepController::GetInputParameters()
 {
   chi::InputParameters params = ChiObject::GetInputParameters();
 
-  params.AddOptionalParameter("initial_dt", 0.01, "Initial timestep to use");
+  params.AddOptionalParameter("dt", 0.01, "Initial timestep to use");
+  params.AddOptionalParameter("time", 0.0, "Initial time");
+  params.AddOptionalParameter(
+    "time_index", 0, "Time index. Useful for output control.");
+  params.AddOptionalParameter("end_time", 1.0, "End time");
+  params.AddOptionalParameter(
+    "max_time_steps",
+    -1,
+    "Maximum number of timesteps to take. A negative number disables this.");
+
+  params.AddOptionalParameter("eps",
+                              1.0e-12,
+                              "General time tolerance. This is used to "
+                              "constrict the dt to the end-time.");
 
   return params;
 }
 
 TimeStepController::TimeStepController(const chi::InputParameters& params)
   : ChiObject(params),
-    current_timestep_size_(params.GetParamValue<double>("initial_dt"))
+    dt_(params.GetParamValue<double>("dt")),
+    time_(params.GetParamValue<double>("time")),
+    t_index_(params.GetParamValue<int>("time_index")),
+    end_time_(params.GetParamValue<double>("end_time")),
+    max_time_steps_(params.GetParamValue<int>("max_time_steps")),
+    general_tolerance_(params.GetParamValue<double>("eps"))
+{
+}
+
+/**Overridable method to get the timestep size.*/
+double TimeStepController::GetTimeStepSize()
+{
+  dt_ = StepSizeLimitedToEndTime();
```

Comment:
An accessor changing a member??? Not a fan.

Reply:
I actually agree on this. The next timestep size should be determined during the Advance. Given that nothing right now uses this and that it is still a work in progress, would you be fine with deferring this till future work. With better examples.

Reply:
I guess. I don't think we should introduce features that are both unused and in a state that needs changing. PRs should not create issues, but rather resolve them.

Reply:
Fixed. Have a look at the new implementation and example usage in the point-reactor module.